text/0000-singleton-removal.md

```diff
@@ -0,0 +1,69 @@
+- Feature Name: `singleton-remove`
+- Start Date: 2021-02-11
+- RFC PR: [RFC#11](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/pull/11)
+- Issue: [Unity-Technologies/com.unity.multiplayer#0000](https://github.com/Unity-Technologies/com.unity.multiplayer/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This change removes the `NetworkingManager.Singleton` static, as well as the majority of other static state in MLAPI. 
+
+# Motivation
+[motivation]: #motivation
+
+The purpose of the change is to allow multiple instances of the MLAPI Networking Stack to co-exist in the same process. This enables:
+- in-process servers. This allows developers to architect single-player and host-mode versions of their multiplayer game identically to the dedicated-server version of the same game.
+- integration testing. Both games that use MLAPI and MLAPI itself can write full server/client tests that run entirely in the scope of a single process. 
+- clients talking to multiple servers simultaneously. This is expected to be a requirement for MMO support. 
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+MLAPI supports having multiple `NetworkingManager`s running concurrently. There are a few things to keep in mind when doing so:
+- The simplest case is to only have one `NetworkingManager` per scene. This is what you do to support an in-process server: load a "server" version of your scene, and then call `StartServer` on the `NetworkingManager` in it. All "server" physics calculations must then be performed via the `PhysicsScene` associated with the server scene.
+- If you have multiple `NetworkingManager`s in one scene (for example, to talk to two different servers at once), then you should take care that all placed `NetworkedObject`s be wired to the `NetworkingManager` they are associated with via a serialized field.
+- `NetworkedObject`s spawned by a `NetworkingManager` will always be bound to the `NetworkingManager` that created them, and be placed in the same scene as their parent `NetworkingManager`.
```

Comment:
I'm a little bit curious about inter-`NetworkManager` transfer of `NetworkObject`/`NetworkBehaviour` — also sharing the same `NetworkObject` across multiple `NetworkManager`s. I'm not asking for a solution here but wanted to call out to spike some interest.

Having said that, more persistent/resilient/reliable network ID generation would greatly help in this area as well.

Reply:
Hmm, the only use-case I can think of for two NetworkManagers sharing the same NetworkObject is pretty advanced: the "seamless world" problem of having two adjacent blocks, managed by two different servers, and supporting a networked object crossing from server A to server B. I agree that would be nice to support at the MLAPI level, and probably would require a globally unique networkID.